jekyll/_cci2/configuration-reference.md

```diff
@@ -1391,6 +1391,139 @@ ignore | N | String, or List of Strings | Either a single tag specifier, or a li
+{:.no_toc}
+Sometimes you may wish to run a job with every combination of arguments _except_
+some value or values. You can use an `exclude` stanza to achieve this:
+
+```yaml
+workflows:
+  workflow:
+    jobs:
+      - build:
+          matrix:
+            parameters:
+              a: [1, 2, 3]
+              b: [4, 5, 6]
+            exclude:
+              - a: 3
+                b: 5
+```
+
+The matrix above would result in 8 jobs being run: every combination of the
+parameters `a` and `b`, excluding `{a: 3, b: 5}`
+
+###### Requiring an entire matrix
+{:.no_toc}
+Another job can `require` an entire matrix (that is, every job in the matrix)
+using its `alias`. The `alias` defaults to the name of the job being invoked.
+```yaml
+workflows:
+  workflow:
+    jobs:
+      - deploy:
+          matrix:
+            parameters:
+              version: ["0.1", "0.2"]
+      - another-job:
+          requires:
+            - deploy
+```
+
+This means that `another-job` will require both `deploy-0.1` and `deploy-0.2` to
+finish before it runs. 
+
+You can also manually set the `alias` to a custom value. For example:
+
+```yaml
+workflows:
+  workflow:
+    jobs:
+      - deploy:
+          matrix:
+            parameters:
+              version: ["0.1", "0.2"]
+            # setting a custom alias
+            alias: my-deploy-matrix
+      - another-job:
+          requires:
+            # using a custom alias
+            - my-deploy-matrix
+```
+
+###### Matrix parameters
+{:.no_toc}
+When running a job in a matrix, other job arguments (such as `requires` or
+`name`) can contain matrix parameters, in the form `<< matrix.[parameter-name] >>`
+For example:
+```yaml
+workflows:
+  workflow:
+    jobs:
+      - build-0.1: ...
+      - build-0.2: ...
+      - deploy:
+          matrix:
+            parameters:
+              version: ["0.1", "0.2"]
+          name: "deploy-<< matrix.version >>"
+          requires:
+            - "build-<< matrix.version >>"
+```
+
+This matrix results in two jobs being run. One, called `deploy-0.1`,
+would require the `build-0.1` job. The other, called `deploy-0.2`, would require
+the `build-0.2` job.
+
+In a typical use case, you may wish to require a single job from a matrix from
+within another matrix. To do so, use matrix parameters to set the `name` of the
+required job, and the `requires` stanza of the dependent job:
+
+```yaml
+workflows:
+  workflow:
+    jobs:
+      - build:
+          # provide a name to each job in this matrix
+          name: "build-<< matrix.os >>-<< matrix.version >>"
+          matrix:
+            parameters:
+              version: ["0.1", "0.2"]
+              os: ["windows", "linux"]
+      - deploy:
+          matrix:
+            parameters:
+              version: ["0.1", "0.2"]
+              os: ["windows", "linux"]
+          requires:
+            # require the above name in each job in this matrix
+            - "build-<< matrix.os >>-<< matrix.version >>"
```

Comment:
Does this mean, if `deploy` requires a single build job, you would use the specific parameters in the requires naming? Like: `requires: -"build-windows-0.2"` for example? Or is this just showing the full `deploy` matrix relies on the full `build` matrix?

Reply:
There are three possible scenarios:
- _each_ deploy job depends on exactly one `build` job. So the `deploy-windows-0.2` job depends on `build-windows-0.2`, `deploy-macos-0.1` depends on `build-macos-0.1`, etc. That's this example ^

- all `deploy` jobs depends on a single `build` job. In that case the `requires` wouldn't have any templating at all, it's just be `requires: [my-job-name]`

- we don't want to run any deploy jobs unless ALL the build jobs succeeded. Then we will require the whole build matrix, maybe using an alias: `requires: [my-build-matrix-alias]`

Does that answer your question?

Reply:
Yes I think so thank you!